x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ClientHelperTests.java

```diff
@@ -442,8 +444,8 @@ public void testGetPersistableSafeSecurityHeaders() throws IOException {
         }
 
         // Rewritten for older version
-        final Version previousVersion = VersionUtils.randomPreviousCompatibleVersion(random(), Version.CURRENT);
-        when(discoveryNodes.getMinNodeVersion()).thenReturn(previousVersion);
+        final TransportVersion previousVersion = TransportVersionUtils.randomCompatibleVersion(random());
+        when(clusterState.getMinTransportVersion()).thenReturn(previousVersion);
```

Comment:
`VersionUtils.randomPreviousCompatibleVersion` returns a version that does not equal to the given version. In this case, it will not return `Version.CURRENT`. But `TransportVersionUtils.randomCompatibleVersion` can return a version that is current. So the version is not always "previous".

Reply:
Oh yes, good spot